packages/application-shell/src/components/navbar/navbar.tsx

```diff
@@ -111,8 +119,59 @@ export const ApplicationMenu = (props: ApplicationMenuProps) => {
   const [submenuVerticalPosition, setSubmenuVerticalPosition] = useState(0);
   const [isSubmenuAboveMenuItem, setIsSubmenuAboveMenuItem] = useState(false);
   const [isSubmenuFocused, setIsSubmenuFocused] = useState(false);
+  const [percentageX, setPercentageX] = useState(0);
+  const [percentageY, setPercentageY] = useState(0);
   const observerRef = useRef<IntersectionObserver | null>(null);
   const submenuRef = useRef<HTMLUListElement>(null);
+  const submenuSafeAreaRef = useRef<HTMLElement>(null);
+
+  /* Getting the width and height of the menu*/
+  const subRefBoundingClientRect = submenuRef.current?.getBoundingClientRect();
+  const { width: menuItemWidth, height: menuItemHeight } =
+    subRefBoundingClientRect! ?? {};
+
+  // /* We want to track the left, top, width, and height of the safe area */
+  const submenuSafeAreaRefBoundingClientRect =
+    submenuSafeAreaRef.current?.getBoundingClientRect();
+  const safeAreaLeftPos = submenuSafeAreaRefBoundingClientRect?.left || 0;
+  const safeAreaTopPos = submenuSafeAreaRefBoundingClientRect?.top || 0;
+  const safeAreaWidth = submenuSafeAreaRefBoundingClientRect?.width || 0;
+  const safeAreaHeight = submenuSafeAreaRefBoundingClientRect?.height || 0;
+
+  const handleMouseMove = useCallback(
+    (e) => {
+      const localX = e.clientX - safeAreaLeftPos;
+      const localY = e.clientY - safeAreaTopPos;
+
+      setPercentageX((localX / safeAreaWidth) * 100);
+      setPercentageY((localY / safeAreaHeight) * 100);
+    },
+    [safeAreaHeight, safeAreaLeftPos, safeAreaTopPos, safeAreaWidth]
+  );
+
+  useEffect(() => {
+    handleMouseMove((e: MouseEventHandler) => e);
+    window.addEventListener('mousemove', handleMouseMove);
+
+    return () => {
+      window.removeEventListener('mousemove', handleMouseMove);
+    };
+  }, [
+    handleMouseMove,
+    menuItemHeight,
+    menuItemWidth,
+    safeAreaHeight,
+    safeAreaLeftPos,
+    safeAreaTopPos,
+    safeAreaWidth,
+  ]);
+
+  useLayoutEffect(() => {
+    submenuRef.current?.style.setProperty(
+      '--safe-start',
+      `${percentageX}% ${percentageY + 1}%`
```

Comment:
Can we also explain why adding 1 is needed?

Reply:
Updated here: https://github.com/commercetools/merchant-center-application-kit/pull/3536/commits/b662e8f9258af8523279fa2a301a35959e6a6b33

I left a comment just to explain it. We want the cursor so be slightly inside the safe area while we move it so we add an extra digit to keep it just inside the triangle at it's apex. Otherwise, it would be exactly at the end of the triangle and that would not shield the cursor from triggering an event we want to prevent.